Documentation/Usage.md

```diff
@@ -397,3 +398,34 @@ manifest API).
 It is *not* expected the packages would ever use this feature unless absolutely
 necessary to support existing clients. In particular, packages *should not*
 adopt this syntax for tagging versions supporting the _latest GM_ Swift version.
+
+### Editable Packages
+
+SwiftPM supports editing dependencies, which means moving it into a location under `Packages/` directory. 
+
+If a such an editable package is present in Packages, then swift build will always use the exact sources in this directory to build, regardless of it's state, git repository status, tags, or the tag desired by dependency resolution. In other words, this will "just build" against the sources that are present.
+
+When an editable package is present, it will be used to satisfy all instances of that Package in the depencency graph. It is possible to edit all, some, or none of the packages in a dependency graph, without restriction.
+
+Editable package are best used to do experimentation with dependency code or create and submit a patch in the dependency owner's repository (upstream).
+
+There are two ways to put a package in editable state:
+
+    $ swift package edit Foo --branch bugFix
+
+This will create a branch called `bugFix` from currently resolved version and put the dependency Foo in Packages/ directory. 
+
+    $ swift package edit Foo --revision 969c6a9
+
+This is similar to previous version except that SwiftPM will leave the dependency at a detched HEAD on the specified revision.
```

Comment:
Is it possible to just do `swift package edit Foo` to leave it at the currently checked out revision, or does a revision or branch always need to be specified?  (it feels like we discussed this when discussing the swift-evolution proposal but I don't remember now where we came down on this)

If one or the other always has to be specified, then perhaps that should be explicitly stated and the rationale given, since it feels very natural to want to just do `swift package edit Foo`.